### PR TITLE
Payment method restriction by countries

### PR DESCRIPTION
**Pull request by DerCalli**
**Description**

- Skeleton

### CODE REVIEW

**Comment by DerCalli** [comment]
Sorry, had to rename the package.

**Review by SpiderDan98 — APPROVED** [review]

File: bundles/payment-country-restriction/phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,10 @@
+parameters:
+    level: 5
+    reportUnmatchedIgnoredErrors: false
+    scanDirectories:
+        - src/Orm/Zed
+    ignoreErrors:
+        - '#.* on an unknown class Generated\\(Client|Glue|Service|Yves|Zed)\\.*#'
+        - '#.* to an undefined method.*Orm\\Zed\\Tax\\Persistence\\SpyTaxRateQuery::useSpyRegionQuery\(\)#'
+        - '#.* \$formData of method FondOfOryx\\Yves\\AvailabilityAlert\\Controller\\SubmitController::registerForAvailabilityAlert\(\) expects array, bool\|float\|int\|string given\.*#'
```
Comment (julianzimmermann) [inline bundles/payment-country-restriction/phpstan.neon:9]: needed?

File: bundles/payment-country-restriction/src/FondOfOryx/Shared/PaymentCountryRestriction/PaymentCountryRestrictionConstants.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace FondOfOryx\Shared\PaymentCountryRestriction;
+
+interface PaymentCountryRestrictionConstants
+{
+    /**
+     * @var string
+     */
+    public const PAYMENT_METHOD_WITH_ALLOWED_COUNTRIES = 'PaymentCountryRestrictionConstants:PAYMENT_METHOD_WITH_ALLOWED_COUNTRIES';
```
Comment (julianzimmermann) [inline bundles/payment-country-restriction/src/FondOfOryx/Shared/PaymentCountryRestriction/PaymentCountryRestrictionConstants.php:10]: remove Constatns

**Review by julianzimmermann — APPROVED** [review]
Will this handle it for all payment methods or only invoice?

File: bundles/payment-country-restriction/composer.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            "name": "Pascal Fischer",
+            "email": "p-fischer@gmx.net"
+        }
+    ],
+    "require": {
+        "php": ">=7.3",
+        "spryker/payment-extension": "^1.0.0"
+    },
+    "require-dev": {
+        "roave/security-advisories": "dev-latest",
```
Comment (daniel-rose) [inline bundles/payment-country-restriction/composer.json:16]: needed?

File: bundles/payment-country-restriction/src/FondOfOryx/Shared/PaymentCountryRestriction/PaymentCountryRestrictionConstants.php (hunk 2 above)
Comment (daniel-rose) [inline bundles/payment-country-restriction/src/FondOfOryx/Shared/PaymentCountryRestriction/PaymentCountryRestrictionConstants.php:10]: Maybe exclusion is better then including?

Comment (daniel-rose) [inline bundles/payment-country-restriction/src/FondOfOryx/Shared/PaymentCountryRestriction/PaymentCountryRestrictionConstants.php:10]:
```php
public const BLACKLISTED_PAYMENT_COUNTRY_COMBINATIONS = 'FOND_OF_ORYX:PAYMENT_COUNTRY_RESTRICTION:BLACKLISTED_PAYMENT_COUNTRY_COMBINATIONS';
```

File: bundles/payment-country-restriction/src/FondOfOryx/Zed/PaymentCountryRestriction/PaymentCountryRestrictionConfig.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+use FondOfOryx\Shared\PaymentCountryRestriction\PaymentCountryRestrictionConstants;
+use Spryker\Zed\Kernel\AbstractBundleConfig;
+
+class PaymentCountryRestrictionConfig extends AbstractBundleConfig
+{
+    /**
+     * @return array<string, string>
+     */
+    public function getPaymentMethodWithAllowedCountries(): array
```
Comment (daniel-rose) [inline bundles/payment-country-restriction/src/FondOfOryx/Zed/PaymentCountryRestriction/PaymentCountryRestrictionConfig.php:13]: Look at the comments for constants class.

File: bundles/payment-country-restriction/src/FondOfOryx/Zed/PaymentCountryRestriction/PaymentCountryRestrictionConfig.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace FondOfOryx\Zed\PaymentCountryRestriction;
+
+use FondOfOryx\Shared\PaymentCountryRestriction\PaymentCountryRestrictionConstants;
+use Spryker\Zed\Kernel\AbstractBundleConfig;
+
+class PaymentCountryRestrictionConfig extends AbstractBundleConfig
+{
+    /**
+     * @return array<string, string>
```
Comment (daniel-rose) [inline bundles/payment-country-restriction/src/FondOfOryx/Zed/PaymentCountryRestriction/PaymentCountryRestrictionConfig.php:11]: Add an example to the phpdoc.

**Review by daniel-rose — APPROVED** [review]

**Comment by DerCalli** [comment]
> Will this handle it for all payment methods or only invoice?

You can specify any payment method that should be excluded via a country restriction

**Pull request merged**